screenpy_playwright/questions/browser_url.py

```diff
@@ -0,0 +1,42 @@
+"""Ask about the current browser URL."""
+
+from screenpy import Actor, UnableToAnswer, beat
+
+from screenpy_playwright.abilities import BrowseTheWebSynchronously
+
+
+class BrowserURL:
+    """Ask about the current browser URL.
+
+    Examples::
+
+        the_actor.should(
+            See.the(BrowserURL(), ReadsExactly("https://www.nintendo.com"))
+        )
+    """
+
+    def describe(self) -> str:
+        """Describe the Question."""
+        return "The current browser URL."
+
+    @beat("{} reads the current browser URL.")
+    def answered_by(self, the_actor: Actor) -> str:
+        """Ask the Actor about the current browser URL."""
+        page = the_actor.ability_to(BrowseTheWebSynchronously).current_page
+        if page is None:
+            msg = (
+                "Actor does not have a current page from which to get the URL."
+                " Did you forget to `Open` a page?"
+            )
+            raise UnableToAnswer(msg)
```

Comment:
We really ought to come up with a way to handle this logic in the ability to avoid having to do this in _every_ action.

Reply:
What if we make a property on each ability that handles this for us?

```python
class BrowseTheWebSynchronously:
    pages: list[Page]

    @property
    def page(self) -> Page:
        if self.current_page is None:
            msg = "No current page found.  Did you forget to `Open` a page?"
            raise AbilityError(msg)
        return self.current_page
```

Reply:
Oh, i think i like that approach, that will make this much easier. Agreed!